src/_pytest/nodes.py

```diff
@@ -102,31 +102,33 @@ class Node(metaclass=NodeMeta):
     Collector subclasses have children; Items are leaf nodes.
     """
 
+    parent: Optional["Node"]
```

Comment:
I *think* mypy should be able to infer this as it's assigned in `__init__` (removing this line should still `reveal_type(self.parent)` -> `Optional[Node]`)

Reply:
this is a side-effect needed primarily due to session also being a node and having parent = None, ideally that ceases to be the case

Reply:
turns out it is, i suspect i let a temporary issue misled me